influence_benchmark/stats/retroactive_evals.py

```diff
@@ -0,0 +1,251 @@
+import asyncio
+import itertools
+import multiprocessing
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List
+
+import pandas as pd
+from tqdm import tqdm
+
+from influence_benchmark.backend.openai_backend import GPTBackend
+from influence_benchmark.environment.assessor_model import AssessorModel
+from influence_benchmark.root import LOADED_DOTENV
+from influence_benchmark.stats.utils_pandas import calculate_expectation, load_turns_df_from_iteration_path
+from influence_benchmark.utils.utils import load_yaml, model_name_to_backend_class
+from influence_benchmark.utils.utils_prob import check_simplex_and_transform
+
+
+@dataclass
+class RetroactiveIterationState:
+    history: List[Dict[str, str]]
+    variables: Dict[str, str]
```

Comment:
This is a bit hacky as the AssessorModel is built to deal with State instances from state.py. But it's probably fine. Maybe worth adding a comment about that